marklogic-data-hub/src/test/java/com/marklogic/hub/dhs/DeployUserAmpsTest.java

```diff
@@ -0,0 +1,137 @@
+package com.marklogic.hub.dhs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.impl.HubConfigImpl;
+import com.marklogic.hub.impl.Versions;
+import com.marklogic.mgmt.ManageClient;
+import com.marklogic.mgmt.ManageConfig;
+import com.marklogic.mgmt.resource.security.AmpManager;
+import com.marklogic.mgmt.resource.security.RoleManager;
+import com.marklogic.mgmt.resource.security.UserManager;
+import com.marklogic.rest.util.ResourcesFragment;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.nio.file.Path;
+
+
+public class DeployUserAmpsTest extends AbstractHubCoreTest {
+    ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void checkIfTestsCanBeRun(){
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion();
+        Assertions.assertTrue(mlVersion.isNightly() || (mlVersion.getMajor() > 10) ||(mlVersion.getMajor() == 10 && mlVersion.getMinor() >= 440));
+    }
+
+    @Test
+    void createAmpWithInheritableRole() {
+        //POST request
+        writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME, "data-hub-developer");
+        ResourcesFragment amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+        Assertions.assertTrue(amps.resourceExists("testAmp"));
+
+        //PUT request
+        writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME,"data-hub-developer", "data-hub-operator");
+
+        runAsDataHubSecurityAdmin().getManageClient().delete("/manage/v2/amps/testAmp?namespace=&document-uri=/data-hub/5/impl/jobs.sjs&modules-database=data-hub-MODULES");
+        amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+        Assertions.assertFalse(amps.resourceExists("testAmp"));
+    }
+
+    @Test
+    void createAmpWithDelegatedUser(){
```

Comment:
This test is failing. I create a custom role based of "data-hub-manage-amps", assign it to a user and use that user to deploy amps. It fails with 403 Forbidden error

Reply:
"data-hub-manage-amps" should be a new role in the project. That way, a dhsa can create a custom role that inherits it. Also, this role will need to inherit the "manage" role so that it has read access to the Manage API.

Reply:
1. "data-hub-manage-amps" is the datahub role with create/remove/get-amp privilege and "admin-database-amp-data-hub-MODULES" granular privilege which is assigned to "data-hub-security-admin". It has been already deployed to the db as part of installation. I am creating a custom role ("delegated-amp-manager") that has "data-hub-manage-amps" (dh-security-admin deploys this role) and create a user with that role that tries to deploy amps. That has failed
2. If you mean "delegated-amp-manager" should have "manage" role in addition to "data-hub-manage-amps" role, "data-hub-security-admin" cannot deploy. It fails as well.